python/pyspark/pandas/namespace.py

```diff
@@ -1049,6 +1049,10 @@ def read_excel(
         Duplicate columns will be specified as 'X', 'X.1', ...'X.N', rather than
         'X'...'X'. Passing in False will cause data to be overwritten if there
         are duplicate names in the columns.
+        .. note:: This process is not case sensitive. If two columns are spelled the
+        same with different casing then an ambiguity error will arise. Specifying
+        `spark.conf.set("spark.sql.caseSensitive","true")` will resolve this issue.
```

Comment:
`spark.sql.caseSensitive` configuration is actually discouraged. I think this is actually a general issue in pandas API on Spark itself instead of this specific API. Should probably write down somewhere like https://spark.apache.org/docs/latest/api/python/user_guide/pandas_on_spark/faq.html